skyportal/handlers/api/user.py

```diff
@@ -19,50 +20,122 @@
     Stream,
 )
 
+from skyportal.model_util import role_acls, all_acl_ids
 
+log = make_log("api/user")
 env, cfg = load_env()
 
 
+def set_default_role(user, session):
+    if (
+        cfg['user.default_role'] is not None
+        and isinstance(cfg['user.default_role'], str)
+        and cfg['user.default_role'] in role_acls
+    ):
+        role = session.query(Role).filter(Role.id == cfg['user.default_role']).first()
+        'finding the role in the database'
+        if role is None:
+            log(
+                f"Invalid default_role configuration value: {cfg['user.default_role']} does not exist"
+            )
+        else:
+            session.add(UserRole(user_id=user.id, role_id=role.id))
+
+
+def set_default_acls(user, session):
+    if cfg['user.default_acls'] is not None:
+        for acl_id in cfg['user.default_acls']:
+            if acl_id not in all_acl_ids:
+                log(
+                    f"Invalid default_acls configuration value: {acl_id} does not exist"
+                )
+        for acl_id in cfg['user.default_acls']:
+            session.add(UserACL(user_id=user.id, acl_id=acl_id))
+
+
+def set_default_group(user, session):
+    default_groups = []
+    if cfg['misc.public_group_name'] is not None:
+        default_groups.append(cfg['misc.public_group_name'])
+    if cfg['user.default_groups'] is not None and isinstance(
+        cfg['user.default_groups'], list
+    ):
+        default_groups.extend(cfg['user.default_groups'])
+    default_groups = list(set(default_groups))
+    for default_group_name in default_groups:
+        group = session.query(Group).filter(Group.name == default_group_name).first()
+        if group is None:
+            log(
+                f"Invalid default_group configuration value: {default_group_name} does not exist"
+            )
+        else:
+            session.add(GroupUser(user_id=user.id, group_id=group.id, admin=True))
+            if group.streams:
+                for stream in group.streams:
+                    session.add(StreamUser(stream_id=stream.id, user_id=user.id))
+
+
 def add_user_and_setup_groups(
     username,
     first_name=None,
     last_name=None,
     contact_phone=None,
     contact_email=None,
     roles=[],
+    acls=[],
     group_ids_and_admin=[],
     oauth_uid=None,
     expiration_date=None,
 ):
-    # Add user
-    user = User(
-        username=username.lower(),
-        role_ids=roles,
-        first_name=first_name,
-        last_name=last_name,
-        contact_phone=contact_phone,
-        contact_email=contact_email,
-        oauth_uid=oauth_uid,
-        expiration_date=expiration_date,
-    )
-    DBSession().add(user)
-    DBSession().flush()
+    with DBSession() as session:
+        try:
+            session.execute(
+                f'LOCK TABLE {User.__tablename__} IN SHARE ROW EXCLUSIVE MODE'
+            )
```

Comment:
If there's an error does this lock ever get released?

Reply:
I followed the same logic as what I found in other handlers. But I'm thinking that this lock might not be necessary in the first place don't you think ?

Reply:
I took the syntax of the photometry handler (that uses with DBSession and stuff) as a model and thats where i saw it and thought it might be necessary. Ill remove it then !